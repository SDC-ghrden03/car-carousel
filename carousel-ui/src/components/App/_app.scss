* {
  box-sizing: border-box;
  margin: 0;
}

.app-container {
  width: 650px;
  position: absolute;
  left: 10%;
  top: 100px;

  h1 {
    text-align: center;
    padding: .5em;
  }
}