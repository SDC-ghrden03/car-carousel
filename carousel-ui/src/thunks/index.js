import { setLoading, setError, setCars } from '../actions';
import { fetchData } from '../utils/fetchData';
import { combineData } from '../utils/combineData';

export const getAllCars = () => {
  const carsUrl = process.env.REACT_APP_BACKEND_URL + '/cars';
  const imagesUrl = process.env.REACT_APP_BACKEND_URL + '/images';
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const allCars = await fetchData(carsUrl);
      const allImages = await fetchData(imagesUrl);
      dispatch(setLoading(false));
      const combinedData = combineData(allCars, allImages);
      dispatch(setCars(combinedData));
    } catch (err) {
      dispatch(setError(err.message));
    }
  }
}