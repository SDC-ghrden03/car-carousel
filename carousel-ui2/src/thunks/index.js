import { setLoading, setError, setCars } from '../actions';
import { fetchData } from '../utils/fetchData';
import { combineData } from '../utils/combineData';


export const getAllCars = () => {
  //need to change this to an EC2 url for the redis caching layer VVVV
  // const carsUrl = `http://ec2-18-144-38-11.us-west-1.compute.amazonaws.com/cars/9287654?url=http://ec2-13-52-75-33.us-west-1.compute.amazonaws.com`;
  // //   const carsUrl = `http://${process.env.REDIS_URL}/cars/9987654?url=http://${process.env.SERVER_URL}`;
  const carsUrl = 'http://localhost:3001/cars/1';
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const allCars = await fetchData(carsUrl);
      // const allImages = await fetchData(imagesUrl);
      dispatch(setLoading(false));
      // const combinedData = combineData(allCars, allImages);
      const combinedData = allCars;
      dispatch(setCars(combinedData));
    } catch (err) {
      dispatch(setError(err.message));
    }
  }
}