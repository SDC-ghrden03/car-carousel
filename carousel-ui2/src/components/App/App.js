import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Route, Switch } from 'react-router-dom';
import { getAllCars } from '../../thunks';
import { Carousel } from '../Carousel';
import { PageNotFound } from '../PageNotFound';
// import './_app.scss';

const App = () => {
  const { cars, loading } = useSelector(state => ({
    cars: state.cars,
    loading: state.loading
  }));

  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getAllCars());
  }, []);

  const findCar = id => {
    const car = cars.find(car => car.id === id);
    return car;
  }
  return (
    <div className='app-container'>
      <Route exact path='/' component={PageNotFound}/>
      {!loading ?
      <Switch>
        <Route exact path='/cars/:id' render={({ match }) => {
          const { id } = match.params;
          const car = findCar(parseInt(id));
          return car ? <Carousel {...car} /> : <PageNotFound />;
        }} />
      </Switch> :
      <h1>Loading...</h1>
    }
    </div>
  );
}

export default App;
