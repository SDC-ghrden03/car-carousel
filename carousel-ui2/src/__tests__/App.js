import React from 'react';
import App from '../components/App/App';
import { shallow } from 'enzyme';
import { getAllCars } from '../thunks';

jest.mock('react-redux', () => ({
  useDispatch: () => { },
  useEffect: () => { },
  useSelector: () => ({
    cars: { 1: 'test' },
    loading: false
  }),
}));

jest.mock('../thunks');

describe('App', () => {
  let wrapper;
  window.fetch = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<App />);
  });

  it('should not contain any errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });


});
