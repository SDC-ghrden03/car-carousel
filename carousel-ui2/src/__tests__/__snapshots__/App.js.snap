// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot 1`] = `
<div
  className="app-container"
>
  <h1>
    Car Carousel
  </h1>
  <Switch>
    <Route
      exact={true}
      path="/cars/:id"
      render={[Function]}
    />
  </Switch>
</div>
`;
